### ISSUE TITLE
building error

### ISSUE THREAD

**Issue by er1c-cartman** [suchetanrs/traversability_mapping#1]
Firstly, I want to express my gratitude for sharing your remarkable work.

I've encountered an issue while attempting to build your package. Despite diligently following your instructions, I encountered an error.

I'm operating within the Ubuntu 20.04 environment with ROS2 Foxy. Could you kindly advise on resolving this error?

Thank you in advance. Best regards.

Install the project...
-- Install configuration: "Release"
CMake Error at cmake_install.cmake:62 (file):
  file cannot create directory: /usr/local/share/sophus/cmake.  Maybe need
  administrative privileges.

make: *** [Makefile:110: install] Error 1
-- Configuring done (0.0s)
-- Generating done (0.0s)
-- Build files have been written to: /home/home/tra_ws/src/traversability_mapping/traversability_mapping_common/build
[ 50%] Building CXX object CMakeFiles/type_conversion.dir/src/type_conversion.cpp.o
In file included from /home/home/tra_ws/src/traversability_mapping/traversability_mapping_common/src/type_conversion.cpp:7:
/home/home/tra_ws/src/traversability_mapping/traversability_mapping_common/include/traversability_mapping_common/type_conversion.hpp:12:10: fatal error: sophus/se3.hpp: No such file or directory
   12 | #include "sophus/se3.hpp"
      |          ^~~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [CMakeFiles/type_conversion.dir/build.make:76: CMakeFiles/type_conversion.dir/src/type_conversion.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:83: CMakeFiles/type_conversion.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
[ 50%] Building CXX object CMakeFiles/type_conversion.dir/src/type_conversion.cpp.o
In file included from /home/home/tra_ws/src/traversability_mapping/traversability_mapping_common/src/type_conversion.cpp:7:
/home/home/tra_ws/src/traversability_mapping/traversability_mapping_common/include/traversability_mapping_common/type_conversion.hpp:12:10: fatal error: sophus/se3.hpp: No such file or directory
   12 | #include "sophus/se3.hpp"
      |          ^~~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [CMakeFiles/type_conversion.dir/build.make:76: CMakeFiles/type_conversion.dir/src/type_conversion.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:83: CMakeFiles/type_conversion.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
Reading package lists... Done
Building dependency tree       
Reading state information... Done
ros-foxy-sensor-msgs is already the newest version (2.0.5-1focal.20230527.050830).
The following packages were automatically installed and are no longer required:
  gconf-service gconf-service-backend gconf2-common libdjvulibre-dev
  libgconf-2-4 liblqr-1-0-dev libmagick++-6-headers libmagick++-6.q16-8
  libmagickcore-6-arch-config libmagickcore-6-headers libmagickwand-6-headers
  librhash0 librsvg2-dev linux-headers-5.15.0-102-generic
  linux-hwe-5.15-headers-5.15.0-102 linux-image-5.15.0-102-generic
  linux-modules-5.15.0-102-generic linux-modules-extra-5.15.0-102-generic
  nsight-compute-2023.3.0
Use 'sudo apt autoremove' to remove them.
0 upgraded, 0 newly installed, 0 to remove and 2 not upgraded.
Reading package lists... Done
Building dependency tree       
Reading state information... Done
E: Unable to locate package ros-foxy-grid-map
Reading package lists... Done
Building dependency tree       
Reading state information... Done
ros-foxy-nav-msgs is already the newest version (2.0.5-1focal.20230527.050829).
The following packages were automatically installed and are no longer required:
  gconf-service gconf-service-backend gconf2-common libdjvulibre-dev
  libgconf-2-4 liblqr-1-0-dev libmagick++-6-headers libmagick++-6.q16-8
  libmagickcore-6-arch-config libmagickcore-6-headers libmagickwand-6-headers
  librhash0 librsvg2-dev linux-headers-5.15.0-102-generic
  linux-hwe-5.15-headers-5.15.0-102 linux-image-5.15.0-102-generic
  linux-modules-5.15.0-102-generic linux-modules-extra-5.15.0-102-generic
  nsight-compute-2023.3.0
Use 'sudo apt autoremove' to remove them.
0 upgraded, 0 newly installed, 0 to remove and 2 not upgraded.
CMake Error at CMakeLists.txt:12 (find_package):
  By not providing "Findtraversability_mapping_common.cmake" in
  CMAKE_MODULE_PATH this project has asked CMake to find a package
  configuration file provided by "traversability_mapping_common", but CMake
  did not find one.

  Could not find a package configuration file provided by
  "traversability_mapping_common" with any of the following names:

    traversability_mapping_commonConfig.cmake
    traversability_mapping_common-config.cmake

  Add the installation prefix of "traversability_mapping_common" to
  CMAKE_PREFIX_PATH or set "traversability_mapping_common_DIR" to a directory
  containing one of the above files.  If "traversability_mapping_common"
  provides a separate development package or SDK, be sure it has been
  installed.

-- Configuring incomplete, errors occurred!
make: *** No targets specified and no makefile found.  Stop.
make: *** No rule to make target 'install'.  Stop.

**Comment by er1c-cartman** [comment]
well I've solved
Is there any example launch file that I could run in your package?

Best regards :)

**Comment by suchetanrs** [comment]
Hi @er1c-cartman , sorry for the delayed reply. Was caught up with something else.
I was able to recreate this issue. Turns out I had not given the right permissions while doing ```make install``` in the ```build.sh``` file. I've updated the file and pushed the latest changes. However, I tested the build on ROS 2 Humble and not on Foxy. 

I am also planning to add a demo this week with a bag file to run this program. Let me know if you have any other issues.
